### PR TITLE
[e2e] Add test case for PVC creation in RKE2 cluster

### LINKED ISSUE #748: [TEST] Add test case for creating PVC in RKE2 cluster
## What's the test to develop? Please describe

https://github.com/harvester/harvester/issues/3647

## Prerequisite and dependency of test

Any prerequisite environment and pre-condition required for this test. 
Provide test case dependency here if any.

- [ ] Write the test skeleton
- [ ] Implement the test of creating PVC in RKE2 cluster
- [ ] Observe the test result on daily running test pipeline

## Additional context

Add any other context or screenshots about the test request here.

### PR DESCRIPTION
**Pull request by lanfon72**
## Changes
- **New** Cluster API for cluster management
    - Add Cluster's PVC Manager
    - Add Cluster's Storage Class Manager
    - Add Cluster's PV Manager
    - Add PVC Spec for PVC creation
- **New** test case to cover https://github.com/harvester/harvester/issues/3647 

To close #748

### CODE REVIEW

**Comment by lanfon72** [comment]
The new test case will depends on previous test case created RKE2 cluster.

File: harvester_e2e_tests/integration/test_rancher_integration.py [hunk 1]
```diff
@@ -280,6 +280,32 @@ def test_create_rke2(self, rancher_api_client, unique_name, harvester_mgmt_clust
...
+        capi = rancher_api_client.clusters.explore(cluster_id)
+
+        spec = capi.pvcs.Spec(1)
+        code, data = capi.pvcs.create(unique_name, spec)
+        assert 201 == code, (code, data)
+
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = capi.pvcs.get(unique_name)
+            if "Bound" == data['status'].get('phase'):
```
Comment (khushboo-rancher) [inline harvester_e2e_tests/integration/test_rancher_integration.py:297]: Can you also add a check for having the right storage class and size?

**Review by khushboo-rancher — CHANGES_REQUESTED** [review]
Can we also check that relevant PV is created with correct info. I understand that if the PVC's status is `bound` that means the PV is created successfully but as a part of testing we should also check the PV's details.

File: apiclient/rancher_api/cluster_models.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    def mode_rwo(self):
+        return bool(self.access_modes & AccessModes.ReadWriteOnce)
+
+    @mode_rwo.setter
+    def mode_rwo(self, enable):
+        op = or_ if enable else xor
+        self.access_modes = op(self.access_modes, AccessModes.ReadWriteOnce)
+
+    @property
+    def mode_rwm(self):
```
Comment (khushboo-rancher) [inline apiclient/rancher_api/cluster_models.py:44]: nit: rwx is general acronym for Read write many.

**Review by khushboo-rancher — APPROVED** [review]

**Comment by khushboo-rancher** [comment]
Thanks @lanfon72, merging this now, you can address [this](https://github.com/harvester/tests/pull/764#discussion_r1169132453) later.

**Pull request merged**